_source/_posts/2023-01-23-keycloak-kubernetes-prod.md

```diff
@@ -0,0 +1,644 @@
+---
+layout: blog_post
+title: "Keycloak in Production with Spring Boot and JHipster"
+author:
+by: contractor
+communities: [devops,security,java]
+description: "A walk-through of building a microservices architecture with JHipster, Keycloak for OAuth 2.0, and Google Kubernetes Engine (GKE) deployment."
+tags: [keycloak, kubernetes, java, jhipster, security, oauth2]
+tweets:
+- ""
+- ""
+- ""
+image:
+type: awareness
+---
+Keycloak is an open-source identity and access management solution that allows you to secure your applications and services with ease. JHipster, on the other hand, is a powerful development platform that provides developers with the tools they need to create modern, scalable, and robust web applications using Spring Boot.
+
+Keycloak is the default OpenID Connect server configured with JHipster. Using Keycloak with JHipster is an excellent way to ensure that your web applications are secure and well-protected from unauthorized access. By integrating Keycloak to your applications, you can easily implement authentication and authorization, manage user identities, and provide secure access to resources.
+
+The JHipster Kubernetes generator creates all the necessary Kubernetes resources, such as deployments, services, and ingresses, based on the configuration of the JHipster application. This includes setting up the database, configuring security, and setting up any necessary environment variables. Since JHipster 8, the Kubernetes sub-generator supports Keycloak for Ingress GKE deployment. In this post, I'll walk you through the generation of a demo Spring Boot microservices application with Keycloak integration using JHipster, and its deployment to Google Kubernetes Engine (GKE).
+
+{% img blog/keycloak-kubernetes-prod/keycloak-gke.png alt:"Keycloak, JHipster, GKE logos" width:"500" %}{: .center-image }
+
```

Comment:
Can you please add a prerequisite section here so folks know what versions to use? For example, I tried Java 20 and got a failure from Gradle.

Reply:
Added a tools section